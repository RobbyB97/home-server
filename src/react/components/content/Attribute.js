import React from 'react'
import {Link, NavLink} from 'react-router-dom'


export default class Attribute extends React.Component {

    render() {
        return (
            <div className={this.props.title ? `attribute ${this.props.title.toLowerCase()}` : "attribute"}>
                <div className="attribute__top">
                    {this.props.image &&
                        <div className="attribute__image">
                            <img src={this.props.image}/>
                        </div>
                    }
                            
                    {this.props.title && 
                        <p className="attribute__title">
                            {this.props.title}
                        </p>
                    }
                </div>

                <div className="attribute__bottom">
                    {this.props.text &&
                        <p className="attribute__text">
                            {this.props.text}
                        </p>
                    }
                </div>
            </div>
        )
    }
}