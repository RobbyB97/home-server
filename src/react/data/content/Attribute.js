export const attribute_fast = {
    image: "/dist/images/icons/speed_icon.png",
    title: "Fast",
    text: "Fully minified files and an optimized structure guarantee your website will always load quickly."
}

export const attribute_responsive = {
    image: "/dist/images/icons/responsive_icon.png",
    title: "Responsive",
    text: "Mobile first development ensures your site works smoothly on all screen sizes."
}

export const attribute_beautiful = {
    image: "/dist/images/icons/paint_icon.png",
    title: "Beautiful",
    text: "A UX focused design process guarantees your website is as functional as it is elegant."
}

export const attribute_accessible = {
    image: "/dist/images/icons/accessibility_icon.png",
    title: "Accessible",
    text: "Compliance with ADA standards ensures your website is accessible and easy to use for everyone."
}
