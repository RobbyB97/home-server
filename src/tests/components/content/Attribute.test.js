import React from 'react'
import {shallow} from 'enzyme'

import Attribute from '../../../react/components/content/Attribute'
import attributes from '../../fixtures/attributes'


test('Render Attribute without props', () => {
    const wrapper = shallow(<Attribute/>)

    expect(wrapper).toMatchSnapshot()
})


test('Render Attribute with all props', () => {
    const wrapper = shallow(
        <Attribute 
            {...attributes[0]}
        />
    )

    expect(wrapper).toMatchSnapshot()
})


test('Render Attribute with some props', () => {
    const wrapper = shallow(
        <Attribute 
            {...attributes[1]}
        />
    )

    expect(wrapper).toMatchSnapshot()
})