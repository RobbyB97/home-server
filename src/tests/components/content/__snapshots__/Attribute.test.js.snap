// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Attribute with all props 1`] = `
<div
  className="attribute title!"
>
  <div
    className="attribute__top"
  >
    <div
      className="attribute__image"
    >
      <img
        src="Image!"
      />
    </div>
    <p
      className="attribute__title"
    >
      Title!
    </p>
  </div>
  <div
    className="attribute__bottom"
  >
    <p
      className="attribute__text"
    >
      Text!
    </p>
  </div>
</div>
`;

exports[`Render Attribute with some props 1`] = `
<div
  className="attribute title!"
>
  <div
    className="attribute__top"
  >
    <p
      className="attribute__title"
    >
      Title!
    </p>
  </div>
  <div
    className="attribute__bottom"
  >
    <p
      className="attribute__text"
    >
      Text!
    </p>
  </div>
</div>
`;

exports[`Render Attribute without props 1`] = `
<div
  className="attribute"
>
  <div
    className="attribute__top"
  />
  <div
    className="attribute__bottom"
  />
</div>
`;
